src/main/kotlin/com/smallcloud/refactai/lsp/LSPProcessHolder.kt

```diff
@@ -436,4 +404,56 @@ class LSPProcessHolder(val project: Project) : Disposable {
             return "<no-address-configured>"
         }
     }
+
+    companion object {
+        val BIN_PATH = Path(
+            getTempDirectory(),
+            ApplicationInfo.getInstance().build.toString().replace(Regex("[^A-Za-z0-9 ]"), "_") +
+                "_refact_lsp${getExeSuffix()}"
+        ).toString()
+
+        @JvmStatic
+        fun getInstance(project: Project): LSPProcessHolder? = project.service()
+
+        var buildInfo: String = ""
+        private val initialized = AtomicBoolean(false)
+        private val logger = Logger.getInstance("LSPProcessHolder")
+
+        // only one time
+        fun initialize() {
+            val shouldInitialize = !initialized.getAndSet(true)
+            if (!shouldInitialize) return
+
+            Companion::class.java.getResourceAsStream(
+                "/bin/${binPrefix}/refact-lsp${getExeSuffix()}"
+            ).use { input ->
+                if (input == null) {
+                    emitError("LSP server is not found for host operating system, please contact support")
+                } else {
+                    for (i in 0..4) {
+                        try {
+                            val path = Paths.get(BIN_PATH)
+                            path.parent.toFile().mkdirs()
+                            Files.copy(input, path, StandardCopyOption.REPLACE_EXISTING)
+                            setExecutable(path.toFile())
+                            break
+                        } catch (e: Exception) {
+                            logger.warn(e.message)
+                        }
+                    }
+                }
+            }
+        }
+
+        fun getCustomizationDirectly(): JsonObject? {
+            val process = GeneralCommandLine(listOf(BIN_PATH, "--print-customization"))
```

Comment:
Causes issues when xDebug is set, 
`/v1/customization` could be used to get the configuration